src/mixins/filing-template-mixin.ts

```diff
@@ -824,7 +824,7 @@ export default class FilingTemplateMixin extends DateMixin {
     })
 
     // restore Restoration data
-    this.mutateRestorationApprovalType(this.getStateFilingRestoration?.approvalType)
+    this.mutateRestorationApprovalType(filing.restoration.approvalType)
```

Comment:
I don't think this'll work! This is since when we create a new filing, it only has a type. We were getting the approval type from state filing on load to know whether we should show the approval type section or not. Also, the approval type is always going to be the same as the state filing in this case. 

For reference, this is what we get when creating a brand new limited to full filing:
![for reference](https://user-images.githubusercontent.com/122301442/231218084-79779f01-a11f-4b8b-9784-9a26b3ba2d57.PNG)

Reply:
Hey @JazzarKarim thanks for spotting this.  I'll make a change so this works.

Reply:
Hey @severinbeauvais I think we made this change together.  I'd like to discuss the issue Karim has raised above to make sure I understand the implications.

Reply:
Unless I'm mistaken, the way it was should be fine which is setting approval type the same as that of state filing. Please correct me if I'm wrong.
CC: @severinbeauvais

Reply:
Fixed by wrapping the action in a conditional.  Now, if the `approvalType` attribute isn't contained in the restoration object, we won't set it.  Thanks @JazzarKarim for spotting that!

Reply:
Awesome! Thanks Jonathan!

I just have one small question. Since we're using the approval type to determine whether we're going to show the approval type section in `ExtendTimeLimit.vue`, will your change have implications on that?

I'm thinking it might since when we create a new limited restoration extension now, the restoration object won't know if it was previously approved by court order or registrar. Correct me if I'm wrong!

Reply:
Nevermind, I just noticed this: https://github.com/bcgov/business-edit-ui/blob/65f9e45b82871a2141f6486500061a3e8e535dd8/src/components/Restoration/ExtendTimeLimit.vue#L93
It should be good! Thanks Jonathan!